### DISCUSSION TITLE
Separate palettes for text and background colours.

### DISCUSSION THREAD

**Discussion by sh1boot** [wezterm/wezterm#6263] [Ideas]
I like my text washed out so I set my palette appropriately, but then those same colours make _terrible_ background colours.  I want background colours which contrast with the text so I can see the text clearly.

I think that rather than fighting the colour configuration of all the different software I'd rather just use a different palette for background colors which guarantees contrast with my foreground colour palette in all combinations (even black on black).

And I guess if I was going to have a dedicated background palette I might also give each colour its own alpha value for added bling.  I don't use transparency or background images myself, but it seems like a logical extension.

Is that a thing?

**Comment by sh1boot** [comment]
Here's a kind of mock-up of what you get with a single palette shared between foreground and background colours, versus what you get by separating them into two palettes:

![mockup](https://www.xn--tkuka-m3a3v.dev/images/twopaletteterminal.png)

The top grid, here, shows what you get when you use the same colour palette for background and text -- when the colour codes 30-37 (text colour) are the same colours as for 40-47 (background colour).  This is the problem.  Many of the combinations are unreadable, because all the colours are intended to contrast with black but they show little contrast with each other.

The middle grid shows the text colours  (30-37) drawing from the same light colour palette, but the background colours (40-47) with the same indices are drawn from a dark palette to ensure contrast.

The bottom grid shows the same as the middle, but inverted.  For those preferring light backgrounds.

I didn't do a great job of levelling out the brightness of the test palette, but you get the idea.

**Comment by sh1boot** [comment]
I just noticed WIndows Terminal has an "adjust indistinguishable colors" option.  Their middle setting is "Only for colors in the color scheme".

https://github.com/microsoft/terminal/blob/-/src/renderer/base/RenderSettings.cpp#L221

Unsurprisingly, it _does_ break block-drawing characters when they need to switch roles between background and foreground colours.

But what else can you do?

**Reply by wez** [reply]
you may want to try the new https://wezterm.org/config/lua/config/text_min_contrast_ratio.html option
